packages/core/contracts/financial-templates/common/FeePayerPoolParty.sol

```diff
@@ -0,0 +1,150 @@
+
+    FeePayerData public feePayerData;
+
+    /****************************************
+     *                EVENTS                *
+     ****************************************/
+
+    event RegularFeesPaid(uint256 indexed regularFee, uint256 indexed lateFee);
+    event FinalFeesPaid(uint256 indexed amount);
+
+    /****************************************
+     *              MODIFIERS               *
+     ****************************************/
+
+    // modifier that calls payRegularFees().
+    modifier fees {
+        payRegularFees();
+        _;
+    }
+    modifier onlyThisContract {
+        require(msg.sender == address(this), "Caller is not this contract");
+        _;
+    }
+
+    /**
+     * @notice Constructs the FeePayer contract. Called by child contracts.
+     * @param _collateralAddress ERC20 token that is used as the underlying collateral for the synthetic.
+     * @param _finderAddress UMA protocol Finder used to discover other protocol contracts.
+     * @param _timerAddress Contract that stores the current time in a testing environment.
+     * Must be set to 0x0 for production environments that use live time.
+     */
+    constructor(
+        address _collateralAddress,
+        address _finderAddress,
+        address _timerAddress
+    ) public Testable(_timerAddress) {
+        feePayerData.collateralCurrency = IERC20(_collateralAddress);
+        feePayerData.finder = FinderInterface(_finderAddress);
+        feePayerData.lastPaymentTime = getCurrentTime();
+        feePayerData.cumulativeFeeMultiplier = FixedPoint.fromUnscaledUint(1);
+    }
+
+    /****************************************
+     *        FEE PAYMENT FUNCTIONS         *
+     ****************************************/
+
+    /**
+     * @notice Pays UMA DVM regular fees (as a % of the collateral pool) to the Store contract.
+     * @dev These must be paid periodically for the life of the contract. If the contract has not paid its regular fee
+     * in a week or more then a late penalty is applied which is sent to the caller. If the amount of
+     * fees owed are greater than the pfc, then this will pay as much as possible from the available collateral.
+     * An event is only fired if the fees charged are greater than 0.
+     * @return totalPaid Amount of collateral that the contract paid (sum of the amount paid to the Store and caller).
+     * This returns 0 and exit early if there is no pfc, fees were already paid during the current block, or the fee rate is 0.
+     */
+    function payRegularFees() public nonReentrant() returns (FixedPoint.Unsigned memory totalPaid) {
+        StoreInterface store = _getStore();
+        uint256 time = getCurrentTime();
+        FixedPoint.Unsigned memory collateralPool = _pfc();
+        totalPaid = feePayerData.payRegularFees(store, time, collateralPool);
+        return totalPaid;
+    }
+
+    function payFinalFees(address payer, FixedPoint.Unsigned memory amount) external onlyThisContract {
```

Comment:
Why make this an external method at all if it can only be called by this contract? Can't you just call the internal method?

Reply:
This is a way to call the pay final fee from the library of Liquidatable...since the library of Liquidtable can not inherit from FeePayer lib (solidity can not support lib inheritance)

Reply:
I see. Makes sense.